### PR TITLE
Making checklist optional

### PR DESCRIPTION
**Pull request by AkshitaB**
Fixes https://github.com/allenai/allennlp/issues/5483

### CODE REVIEW

**Comment by dirkgr** [comment]
@AkshitaB, are you still working on this?

File: README.md [hunk 1]
```diff
@@ -229,6 +231,12 @@ Installing the library and dependencies is simple using `pip`.
 pip install allennlp
 ```
 
+To install the optional dependencies, run
+
+```bash
+pip install allennlp[checklist]
+```
+
```
Comment (epwalsh) [inline README.md:239]:
```suggestion
Or you can just install all optional dependencies with `pip install allennlp[all]`.

```

**Review by epwalsh — APPROVED** [review]
LGTM!

**Comment by h-vetinari** [comment]
While refactoring the dependencies from `setup.py` to `requirements.txt` (and making `checklist` optional), two additional requirements were added:
```
datasets>=1.2.1,<2.0
sqlitedict
```

I noticed this in https://github.com/conda-forge/allennlp-feedstock/pull/35 - just checking that this was intentional and not a C&P error or similar.

**Comment by dirkgr** [comment]
I think you are right. This must have happened as part of a merge? If the tests for https://github.com/allenai/allennlp/pull/5593 pass, that's confirmation enough for me that we don't need these.

**Pull request merged**